lib/logstash/plugin_mixins/http_client.rb

```diff
@@ -7,203 +7,230 @@
+        pool_max: @pool_max,
+        pool_max_per_route: @pool_max_per_route,
+        cookies: @cookies,
+        keepalive: @keepalive
+      }
+
+      if @proxy
+        # Symbolize keys if necessary
+        c[:proxy] = @proxy.is_a?(Hash) ?
+                      @proxy.reduce({}) {|memo,(k,v)| memo[k.to_sym] = v; memo} :
+                      @proxy
       end
 
-      # Symbolize keys if necessary
-      c[:auth] = {
-        :user => @user,
-        :password => @password.value,
-        :eager => true
-      }
-    end
+      if @user
+        if !@password || !@password.value
+          raise ::LogStash::ConfigurationError, "User '#{@user}' specified without password!"
+        end
+
+        # Symbolize keys if necessary
+        c[:auth] = {
+          :user => @user,
+          :password => @password.value,
+          :eager => true
+        }
+      end
 
-    c[:ssl] = {}
-    if @cacert
-      c[:ssl][:ca_file] = @cacert
+      c[:ssl] = ssl_options
+
+      c
     end
 
-    if @truststore
-      c[:ssl].merge!(
-        :truststore => @truststore,
-        :truststore_type => @truststore_type,
-        :truststore_password => @truststore_password.value
-      )
-      
-      if c[:ssl][:truststore_password].nil?
-        raise LogStash::ConfigurationError, "Truststore declared without a password! This is not valid, please set the 'truststore_password' option"
+    private
+
+    def ssl_options
+
+      options = {}
+      if @ssl_certificate_authorities&.any?
+        raise LogStash::ConfigurationError, 'Multiple values on `ssl_certificate_authorities` are not supported by this plugin' if @ssl_certificate_authorities.size > 1
+
+        options[:ca_file] = @ssl_certificate_authorities.first
       end
-    end
 
-    if @keystore
-      c[:ssl].merge!(
-        :keystore => @keystore,
-        :keystore_type => @keystore_type,
-        :keystore_password => @keystore_password.value
-      )
+      if @ssl_truststore_path
+        options[:truststore] = @ssl_truststore_path
+        options[:truststore_type] = @ssl_truststore_type if @ssl_truststore_type
+        options[:truststore_password] = @ssl_truststore_password.value if @ssl_truststore_password
+      elsif @ssl_truststore_password
+        fail LogStash::ConfigurationError, "truststore_password requires ssl_truststore_path you fool"
+      end
 
-      if c[:ssl][:keystore_password].nil?
-        raise LogStash::ConfigurationError, "Keystore declared without a password! This is not valid, please set the 'keystore_password' option"
+      if @ssl_keystore_path
+        options[:keystore] = @ssl_keystore_path
+        options[:keystore_type] = @ssl_keystore_type if @ssl_keystore_type
+        options[:keystore_password] = @ssl_keystore_password.value if @ssl_keystore_password
+      elsif @ssl_keystore_password
+        fail LogStash::ConfigurationError, "ssl_keystore_password requires ssl_keystore_path you fool"
```

Comment:
In the deprecated path, we require providing a `keystore_password` when a `keystore` is provided. I think that needs to be mirrored here and in the matching `ssl_truststore_path` counterpart:

```suggestion
        fail(LogStash::ConfigurationError, "When `ssl_keystore_path` is provided, `ssl_keystore_password` must also be provided") unless @ssl_keystore_password
        options[:keystore] = @ssl_keystore_path
        options[:keystore_type] = @ssl_keystore_type if @ssl_keystore_type
        options[:keystore_password] = @ssl_keystore_password.value if @ssl_keystore_password
      elsif @ssl_keystore_password
        fail LogStash::ConfigurationError, "An `ssl_keystore_password` is not allowed unless `ssl_keystore_path` is provided"
```

Reply:
I think we should not enforce the password. Although it's not recommended, It's possible to create keystore/truststore with empty passwords. If I'm not mistaken, Elasticserach also does not enforce it.